### PR TITLE
RFC: provide a function to generate a contraction plan, and let contraction use it

### PR DESCRIPTION
**Pull request by kshyatt**
The idea is to let people take advantage of CUTENSOR's plan generation and reuse, if they want. If they don't provide a plan, CUTENSOR will just make one for them.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -38,7 +38,7 @@ device!(pick.dev)
 CuArrays.allowscalar(false)
 
 CuArrays.enable_timings()
-
+#=
```
Comment (maleadt) [inline test/runtests.jl]: I think we need to support arguments to `runtests.jl` a la base for our sanity :-)

Comment (kshyatt) [reply test/runtests.jl]: oh my god derp

Comment (kshyatt) [reply test/runtests.jl]: ban me

**Comment by kshyatt** [comment]
OK, tests look happier now. Do you have thoughts on the design here? I know I need to write a test for this use case.

**Comment by kshyatt** [comment]
OK, we have tests and all that. Thoughts?

File: src/tensor/wrappers.jl [hunk 2]
```diff
@@ -239,25 +240,73 @@ function contraction!(
...
+            end
+            cutensorContraction(handle(), plan_ref,
+                                T[convert(T, alpha)], A, B,
+                                T[convert(T, beta)],  C, C,
                                 workspace, sizeof(workspace), stream)
         end
     return C
 end
 
+function contraction_plan(
```
Comment (maleadt) [inline src/tensor/wrappers.jl]: maybe `plan_contraction` re. FFTW?

**Comment by kshyatt** [comment]
Well, the test seems to be passing. I'll change the name as you suggested.

**Comment by kshyatt** [comment]
Tests passed, going to merge.

**Pull request merged**